src/platform/backends/shared/base_snapshot.cpp

```diff
@@ -166,12 +166,12 @@ mp::BaseSnapshot::BaseSnapshot(const std::string& name,
     assert(index > 0 && "snapshot indices need to start at 1");
 }
 
-mp::BaseSnapshot::BaseSnapshot(const QString& filename, VirtualMachine& vm)
-    : BaseSnapshot{read_snapshot_json(filename), vm}
+mp::BaseSnapshot::BaseSnapshot(const QString& filename, VirtualMachine& vm, VirtualMachineDescription& desc)
+    : BaseSnapshot{read_snapshot_json(filename), vm, desc}
 {
 }
 
-mp::BaseSnapshot::BaseSnapshot(const QJsonObject& json, VirtualMachine& vm)
+mp::BaseSnapshot::BaseSnapshot(const QJsonObject& json, VirtualMachine& vm, VirtualMachineDescription& desc)
```

Comment:
You should be able to make the new parameter `const`.

Reply:
Right, done.